html/includes/api_functions.inc.php

```diff
@@ -1319,33 +1404,35 @@ function list_bills()
     }
 
     foreach (dbFetchRows("SELECT `bills`.*,COUNT(port_id) AS `ports_total` FROM `bills` LEFT JOIN `bill_ports` ON `bill_ports`.`bill_id`=`bills`.`bill_id` $sql GROUP BY `bill_name`,`bill_ref` ORDER BY `bill_name`", $param) as $bill) {
-        $rate_data    = $bill;
-        $allowed = '';
-        $used = '';
-        $percent = '';
-        $overuse = '';
-
-        if ($bill['bill_type'] == "cdr") {
-            $allowed = format_si($bill['bill_cdr'])."bps";
-            $used    = format_si($rate_data['rate_95th'])."bps";
-            $percent = round(($rate_data['rate_95th'] / $bill['bill_cdr']) * 100, 2);
-            $overuse = $rate_data['rate_95th'] - $bill['bill_cdr'];
-            $overuse = (($overuse <= 0) ? "-" : format_si($overuse));
-        } elseif ($bill['bill_type'] == "quota") {
-            $allowed = format_bytes_billing($bill['bill_quota']);
-            $used    = format_bytes_billing($rate_data['total_data']);
-            $percent = round(($rate_data['total_data'] / ($bill['bill_quota'])) * 100, 2);
-            $overuse = $rate_data['total_data'] - $bill['bill_quota'];
-            $overuse = (($overuse <= 0) ? "-" : format_bytes_billing($overuse));
-        }
-        $bill['allowed'] = $allowed;
-        $bill['used'] = $used;
-        $bill['percent'] = $percent;
-        $bill['overuse'] = $overuse;
+        if (bill_permitted($bill['bill_id'])) {
```

Comment:
This could be "if !bill_permitted then continue" instead, to avoid a bit of indentation.

Reply:
Updated, but the style rules require this is still inside braces.